lib/OpenQA/WebAPI/Controller/API/V1/User.pm

```diff
@@ -0,0 +1,50 @@
+package OpenQA::WebAPI::Controller::API::V1::User;
+use Mojo::Base 'Mojolicious::Controller';
+
+=pod
+
+=head1 NAME
+
+OpenQA::WebAPI::Controller::API::V1::User
+
+=head1 SYNOPSIS
+
+  use OpenQA::WebAPI::Controller::API::V1::User;
+
+=head1 DESCRIPTION
+
+Implements User API.
+
+=head1 METHODS
+
+=over 4
+
+=item delete()
+
+Deletes an existing user.
+
+=back
+
+=cut
+
+sub delete {
+    my ($self) = @_;
+
+    return $self->render(status => 401) unless $self->current_user->is_admin == 1;
```

Comment:
So it didn't work to use Mojolicious's routes to ensure the user is an admin?

Reply:
I would recommend TDD here, that is to say write a unit test that attempts to delete a user. Once as an admin, and another variant as a regular user. Then drop this line and see what happens. If the route is defined *correctly* in `WebAPI.pm` it should not be needed - you can also play with the routes and follow the code paths to get a better feel for it, if `$api_ra` is not very clear.

Reply:
Good comment. This reminds me of

* http://www.extremeprogramming.org/rules/early.html
* http://www.extremeprogramming.org/rules/simple.html
* http://www.extremeprogramming.org/rules/testfirst.html

I am happy to talk more about XP if there is interest :)

Reply:
Done